packages/connector-schemas/src/error.ts

```diff
@@ -0,0 +1,24 @@
+export enum ConnectorErrorCodes {
+  General,
+  InsufficientRequestParameters,
+  InvalidConfig,
+  InvalidResponse,
+  TemplateNotFound,
+  NotImplemented,
+  SocialAuthCodeInvalid,
+  SocialAccessTokenInvalid,
+  SocialIdTokenInvalid,
+  AuthorizationFailed,
+}
+
+export class ConnectorError extends Error {
+  public code: ConnectorErrorCodes;
+  public data: unknown;
+
+  constructor(code: ConnectorErrorCodes, data?: unknown) {
+    const message = typeof data === 'string' ? data : 'Connector error occurred.';
```

Comment:
I don't think we need this logic. You can just define a key-message mapping.

Reply:
updated.